DataFormats/HcalDigi/interface/QIE11DataFrame.h

```diff
@@ -0,0 +1,56 @@
+#ifndef DATAFORMATS_HCALDIGI_QIE11DATAFRAME_H
+#define DATAFORMATS_HCALDIGI_QIE11DATAFRAME_H
+
+#include "DataFormats/HcalDetId/interface/HcalDetId.h"
+#include "DataFormats/Common/interface/DataFrame.h"
+#include <ostream>
+
+/** Precision readout digi from QIE11 including TDC information
+
+ */
+class QIE11DataFrame : protected edm::DataFrame {
+public:
+
+  static const int WORDS_PER_SAMPLE = 1;
+  static const int HEADER_WORDS = 1;
+
+  QIE11DataFrame() { }
+  QIE11DataFrame(const edm::DataFrameContainer& c, edm::DataFrame::size_type i) : edm::DataFrame(c,i) { }
+  QIE11DataFrame(edm::DataFrame df) : edm::DataFrame(df) { }
+
+  class Sample {
+  public:
+    Sample(const edm::DataFrame& frame, edm::DataFrame::size_type i) : frame_(frame),i_(i) { }
+    int adc() const { return frame_[i_]&0xFF; }
+    int tdc() const { return (frame_[i_]>>8)&0x3F; }
+    bool soi() const { return frame_[i_]&0x4000; }
+    int capid() const { return ((((frame_[0]>>8)&0x3)+i_)%3); }
```

Comment:
considering the bugfix with %3 and %4, I just wanted to check if this one is correct